### ISSUE TITLE
Variable in reduce() causes type change

### ISSUE THREAD

**Issue by ljhhuxiaoba** [neo4j/neo4j#13155]
I execute a query on my neo4j database ,and an error occurs.

Neo4j version: 5.6.0 Enterprise
operating system: windows 10
API：Cypher version 9
query: create b=()-[:T]->() with * unwind nodes(b) as p return 1 limit reduce(a=1,b in []|1)
error message:
Neo.ClientError.Statement.TypeError
Invalid input for function 'nodes()': Expected null to be a path

If I execute another query:create b=()-[:T]->() with * unwind nodes(b) as p return reduce(a=1,b in []|1),then it can run successfully. 
I think the variable "b" defined in reduce() shouldn't make changes to externally defined variable "b".
The same error also occurs in skip clause.

Expected behavior:
The query should run successfully and return 1.

Actual behavior:
The query failed.

**Comment by loveleif** [comment]
Thanks again for the reports @ljhhuxiaoba! I suspect that the query is not semantically correct, or in a grey area at least.

> LIMIT accepts any expression that evaluates to a positive integer — however the expression cannot refer to nodes or relationships.

Here we have:

```cypher
create b=()-[:T]->() 
with * 
unwind nodes(b) as p 
return 1 
limit reduce(a=1,b in []|1)
```

But even if I'm correct, the error message is unexpected. I've created a ticket in our internal bug tracker for further investigation ([link](https://trello.com/c/lwgEMgmj)).

**Comment by Lojjs** [comment]
@ljhhuxiaoba @loveleif  The query (although a bit edge-case) is actually correct. The bug was related to a scoping issue with SKIP and LIMIT and has now been fixed. The fix will be included in upcoming Neo4j 5.11 and 4.4.24 releases. 

Best regards, Louise, Neo4j Cypher team
